witness/src/Dscp/Witness/Workers/Worker.hs

```diff
@@ -91,9 +91,11 @@ blockUpdateWorker =
            | otherwise -> logDebug "blockUpdateWorker: block is useless"
 
     applyNewBlock block = do
+        params <- view (lensOf @WitnessParams)
         logDebug $ "Block " +| hashF (headerHash block) |+
                   " is a direct continuation of our tip, applying"
-        proof <- applyBlock block
+        proof <- maybe id (reportTime "disciplina.timer.block_apply") (wpMetricsEndpoint params) $
```

Comment:
You can create a wrapper that uses the fact we have `Maybe Endpoint` in the context to avoid this `maybe id`.